File: graceful-florist-services/src/main/java/id/vn/thongdanghoang/graceful/dtos/admin/ReportDTO.java
package id.vn.thongdanghoang.graceful.dtos.admin;

import lombok.Builder;

import java.util.List;

@Builder()
public record ReportDTO(
        long totalRevenue,
        long totalRevenueLastWeek,
        long totalOrders,
        long totalOrdersLastWeek,
        long totalOrdersPendingLastWeek,
        long totalOrdersPending,
        List<Long> thisMonthRevenue,
        List<Long> thisYearRevenue
) {
}

File: graceful-florist-web/src/app/modules/admin/model/report.dto.ts
export interface ReportDTO {
  totalRevenue: number;
  totalRevenueLastWeek: number;
  totalOrders: number;
  totalOrdersLastWeek: number;
  totalOrdersPendingLastWeek: number;
  totalOrdersPending: number;
  thisMonthRevenue: number[];
  thisYearRevenue: number[];
}
